src/Editor.js

```diff
@@ -307,11 +374,20 @@ const Editor = () => {
       );
       break;
     case "drag":
+      const gradient =
+        and(keys, ["ShiftLeft", "ShiftRight"]).size > 0 &&
+        (mouse.y - mouse.startY) / (mouse.x - mouse.startX);
```

Comment:
Just a thought, another way of tackling this could be something along these lines:

```js
const lockedAxis =
  and(keys, ["ShiftLeft", "ShiftRight"]).size === 0
    ? null
    : Math.abs(mouse.x - mouse.startX) > Math.abs(mouse.y - mouse.startY)
    ? "x"
    : "y";
```

Reply:
That is much cleaner. I like it